### PR TITLE
Add RFC-002: CORS and authn for Go

### PR DESCRIPTION
**Pull request by akshayjshah**
Add an RFC for creating CORS and authentication packages for users of
Connect in Go.

### CODE REVIEW

**Comment by akshayjshah** [comment]
I've added the current maintainers of the various Connect implementations as reviewers.

I'm formally in favor of adding these small quality-of-life packages to the Connect project. Formally approving this RFC requires approval from an additional 5 maintainers (since we have a total of 10). Maintainers, please indicate support or opposition using Github's code review!

File: docs/governance/rfc/002-go-cors-authn.md
Comment (mattrobenolt) [inline docs/governance/rfc/002-go-cors-authn.md:39]: Would agree.

Do we intend to provide some built-in implementations for things like Basic auth, or mTLS, etc? Or is the goal to just provide the abstraction to implement their own?

**Comment by mattrobenolt** [comment]
Overall, these make sense to me as something to provide as first class support. They are pretty common patterns by most applications. I think having these too will help guide users of Connect on the patterns for implementing more custom things that fall outside of interceptors too.

**Review by mattrobenolt — APPROVED** [review]
Overall, these make sense to me as something to provide as first class support. They are pretty common patterns by most applications. I think having these too will help guide users of Connect on the patterns for implementing more custom things that fall outside of interceptors too.

File: docs/governance/rfc/002-go-cors-authn.md
Comment (pkwarren) [inline docs/governance/rfc/002-go-cors-authn.md:17]: Small suggestion but we should try to replace `allowlist` as a verb here - I think CORS requires servers to send additional HTTP request/response headers (and support preflight requests but that wouldn't affect Connect that uses GET/POST).

**Review by pkwarren — APPROVED** [review]
Very excited to have both of these packages available!

**Review by bufdev — APPROVED** [review]

**Review by smaye81 — APPROVED** [review]

**Review by timostamm — APPROVED** [review]
Providing a source of truth for CORS setup, and a standardized mechanism for propagating identity, are both great additions in my book. 

👍 for adding the features as separate modules. This should keep connect-go as slim and flexible as possible.

**Review by jhump — APPROVED** [review]

File: docs/governance/rfc/002-go-cors-authn.md
Comment (akshayjshah) [reply docs/governance/rfc/002-go-cors-authn.md:10]: I'd expect that adding application-specific request and response headers to the allowlists is fairly common - what do you think?

File: docs/governance/rfc/002-go-cors-authn.md
Comment (akshayjshah) [reply docs/governance/rfc/002-go-cors-authn.md:39]: Just an abstraction to implement their own. We can always add more functionality later on :)

File: docs/governance/rfc/002-go-cors-authn.md
Comment (mattrobenolt) [reply docs/governance/rfc/002-go-cors-authn.md:10]: Yeah, agree. I think even if we had an opinionated default, an additional middleware or whatever can easily append more. But we can hash this out in implementation.

File: docs/governance/rfc/002-go-cors-authn.md
Comment (akshayjshah) [reply docs/governance/rfc/002-go-cors-authn.md:17]: I'll reword. I didn't want to get into the exact CORS flow, but in practice Connect servers need to respond to preflight requests with request and response header allowlist. I'll clarify the text.

File: docs/governance/rfc/002-go-cors-authn.md
Comment (mattrobenolt) [reply docs/governance/rfc/002-go-cors-authn.md:20]: Thanks, yeah, sounded a bit confusing to me even rereading it.

**Review by pkwarren — APPROVED** [review]

**Review by srikrsna-buf — APPROVED** [review]

**Pull request merged**
